cocos/platform/android/libcocos2dx-with-controller/build.gradle

```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 22
-    buildToolsVersion '25.0.0'
+    compileSdkVersion PROP_SDK_VERSION.toInteger()
+    buildToolsVersion '27.0.1'
 
     defaultConfig {
-        minSdkVersion 10
-        targetSdkVersion PROP_TARGET_SDK_VERSION
```

Comment:
So don't we need to distinguish target SDK version and compile SDK version? Why?

Reply:
`targetSdkVersion` and `compileSdkVersion` should both be targeting the latest api.  - https://medium.com/google-developers/picking-your-compilesdkversion-minsdkversion-targetsdkversion-a098a0341ebd.

Doing it like this simplifies things for users who don't know much about android

Reply:
Sounds reasonable.

Reply:
I believe that we should distinguish target SDK version and compile SDK version. If you set target SDK version to >= 23 , then you should deal with runtime permissions yourself.

https://developer.android.com/training/permissions/requesting.html

@dumganhar Please take a look!

Reply:
@Priority10 According my investigation minSdkVersion (lowest possible) <= targetSdkVersion == compileSdkVersion (latest SDK)
http://discuss.cocos2d-x.org/t/android-compile-sdk-version-min-sdk-version-target-sdk-version/39495

Reply:
Have you even read the link?